examples/cpp/xpress_use.cc

```diff
@@ -0,0 +1,101 @@
+
+/**
+ * This method shows two ways to initialize a Xpress solver instance.
+ * Two environment variables are used to specify the Xpress installation paths:
+ *  * XPRESSDIR : Path to the Xpress root directory (containing bin and lib folders)
+ *  * XPRESS : Path to the directory containing Xpress license
+ */
+void useXpressSolver(bool solveAsMip, bool useFactory) {
+  std::unique_ptr<MPSolver> solver = nullptr;
+  if (useFactory) {
+    /* This is the preferred way as the program won't stop if anything went
+       wrong. In such a case, `solver` will take value `nullptr` */
+    std::string xpressName = (solveAsMip ? "XPRESS" : "XPRESS_LP");
+    solver.reset(MPSolver::CreateSolver(xpressName));
+  } else {
+    MPSolver::OptimizationProblemType problemType = (solveAsMip ?
+                                            MPSolver::XPRESS_MIXED_INTEGER_PROGRAMMING
+                                            : MPSolver::XPRESS_LINEAR_PROGRAMMING);
+    /* MPSolver::SupportsProblemType(problem_type) will test if Xpress is
+       correctly loaded and has a valid license. This check is important to keep
+       the program running if Xpress is not correctly installed. With the
+       constructor usage, if Xpress is badly loaded or if there is a problem
+       with the license, the program will abort (SIGABRT)
+    */
+    if (MPSolver::SupportsProblemType(problemType)) {
+        solver.reset(new MPSolver("IntegerProgrammingExample", problemType));
+    }
+  }
+  if (solver == nullptr) {
+       LOG(INFO) << "Xpress solver is not available";
+       return;
+  }
+  // Use the solver
+  /*
+    max -100 x1 + 10 x2
+    s.t. x2 <= 20 x1;
+         30 x1 + 3.5 x2 <= 350
+         0 <= x1 <= 5
+         0 <= x2
+   */
+  const double infinity = solver->infinity();
+  MPVariable* const x1 = solver->MakeIntVar(0, 5, "x1");
+  MPVariable* const x2 = solver->MakeNumVar(0.0, infinity, "x2");
+
+  MPObjective* const objective = solver->MutableObjective();
+  objective->SetCoefficient(x1, -100);
+  objective->SetCoefficient(x2, 10);
+  objective->SetMaximization();
+
+  MPConstraint* const c0 = solver->MakeRowConstraint(-infinity, 0.0);
+  c0->SetCoefficient(x1, -20.0);
+  c0->SetCoefficient(x2, 1);
+
+  MPConstraint* const c1 = solver->MakeRowConstraint(-infinity, 350.0);
+  c1->SetCoefficient(x1, 30.0);
+  c1->SetCoefficient(x2, 3.5);
+
+
+  const MPSolver::ResultStatus result_status = solver->Solve();
+
+  // Check that the problem has an optimal solution.
+  if (result_status != MPSolver::OPTIMAL) {
+    LOG(FATAL) << "Solver returned with non-optimal status.";
+  } else {
+    LOG(INFO) << "Optimal solution found: obj=" << objective->Value();
+  }
+}
+
+
+int main(int argc, char** argv) {
+  absl::SetFlag(&FLAGS_alsologtostderr, true);
+  google::InitGoogleLogging(argv[0]);
+  absl::ParseCommandLine(argc, argv);
+  useXpressSolver(true, true);
+  useXpressSolver(true, false);
+  useXpressSolver(false, true);
+  useXpressSolver(false, false);
```

Comment:
Maybe overkill for an example ? It could confuse a user to think those enums are available in ORTools ? The following?
```cpp
for (bool solveAsMip: {true, false}) {
    for (bool useFactory: {true, false}) {
        useXpressSolver(solveAsMip, useFactory);
    }
}
```

Reply:
Yes, that's good for me !